### PR TITLE
kde/plasma/systemsettings: add some dependencies 

### PR DESCRIPTION
**Pull request by wrvsrx**
## Description of changes

Current some dependencies is missing from kde 6 packaging. So some kde packages work normaly on plasma but fail without plasma. This PR fix dolphin, systemsettings and kcmutils.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/kde/plasma/systemsettings/default.nix [hunk 1]
```diff
@@ -1,4 +1,16 @@
...
+  kdeclarative,
+  ksvg,
+  plasma-workspace,
+}:
 mkKdeDerivation {
   pname = "systemsettings";
+  extraBuildInputs = [
+    knewstuff
+    kdeclarative
+    ksvg
```
Comment (K900) [inline pkgs/kde/plasma/systemsettings/default.nix:14]: Are you sure ksvg is needed?

**Comment by K900** [comment]
The systemsettings changes look OK, the rest I'm less confident about - you'd have to add qtsvg to _every app_ that wants to render SVG icons, which is probably the wrong solution. We may want to somehow propagate that information from the theme instead, and add qtsvg to the global environment.

File: pkgs/kde/plasma/systemsettings/default.nix (hunk 1 above)
Comment (wrvsrx) [reply pkgs/kde/plasma/systemsettings/default.nix]: My mistake, I have removed it.

**Comment by wrvsrx** [comment]
Now this PR only add dependencies to systemsettings.

**Comment by wrvsrx** [comment]
Maybe we do need to add `qtsvg` to all apps that need to render svg icons? We can add qtsvg as an optional dependency that can be overrided by users who don't use svg icons. Adding all dependencies needed is a more "nix" way.

Otherwise users outside plasma need to manually install qtsvg to global scope when they want to use custom svg icons (Maybe they don't want to use themes packaged by nixpkgs).

**Comment by K900** [comment]
> Maybe we do need to add `qtsvg` to all apps that need to render svg icons? We can add qtsvg as an optional dependency that can be overrided by users who don't use svg icons. Adding all dependencies needed is a more "nix" way.
> 
> Otherwise users outside plasma need to manually install qtsvg to global scope when they want to use custom svg icons (Maybe they don't want to use themes packaged by nixpkgs). 

The thing is, "all apps that need to render SVG icons" is just _all apps_. Every single one.

**Comment by wrvsrx** [comment]
Whether we add qtsvg to qt6 apps is a question that we can discuss later. Would you consider approving this PR?

**Comment by K900** [comment]
Actually, looking at it more, these should not be in `systemsettings` after all, they should ideally be propagated somehow from whatever is actually providing those KCMs.

**Comment by FlorianFranzen** [comment]
@wrvsrx  So now it is not fixed at all?

**Comment by wrvsrx** [comment]
@FlorianFranzen No. You still need to add some qt libraries globally to make qt apps work well.

**Pull request closed without merging**